### DISCUSSION TITLE
Typing reusable render function in Vue, storybook v7

### DISCUSSION THREAD

**Discussion by josip2312** [storybookjs/storybook#22157] [Documentation]
There is no example in the documentation on how to create a reusable render function with the CSF 3 syntax -> https://storybook.js.org/docs/vue/writing-stories/args

Example of what I mean: 

```ts
const meta: Meta<typeof FieldAutocomplete> = {...}
export default meta;

type Story = StoryObj<typeof FieldAutocomplete>;

const render = (args) => ({
  components: { FieldAutocomplete },
  setup() {
    const searchTerm = ref();

    return { args, searchTerm };
  },
  template: `
    {{ searchTerm }}
    <FieldAutocomplete v-model="searchTerm" v-bind="args" />
    `,
});

export const Default: Story = {
  render,
};

export const WithLabel: Story = {
  render,

  args: {
    label: "Label here",
  },
};
```

If I do this I will get proper typing for the `args`, but not sure how to type the default render function `args` parameter

![Screenshot 2023-04-19 at 10 04 43](https://user-images.githubusercontent.com/49723492/233010003-7a2a47d9-d4e9-4938-86be-36808511eb2b.png)

**Comment by TomDeSmet** [comment]
A very late answer here, but I had the exact same problem and what I did was add the render function to the Default story and in all my other stories use: `{ render: Default.render }`.
